.github/workflows/build.yml

```diff
@@ -60,27 +59,62 @@ jobs:
         with:
           name: actual-web
           path: packages/desktop-client/build
+      - name: Upload Build Stats
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-stats
+          path: packages/desktop-client/build-stats
 
-  electron:
-    # As electron builds take longer, we only run them in master.
-    strategy:
-      matrix:
-        os:
-          - ubuntu-latest
-          - windows-latest
-          - macos-latest
-    runs-on: ${{ matrix.os }}
+  size-compare:
+    runs-on: ubuntu-latest
+    needs: [web]
+    permissions:
+      pull-requests: write
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up environment
-        uses: ./.github/actions/setup
-      - name: Build Electron
-        run: ./bin/package-electron
-      - name: Upload Build
-        uses: actions/upload-artifact@v3
+      - name: Wait for ${{github.base_ref}} build to succeed
+        uses: fountainhead/action-wait-for-check@v1.1.0
+        id: master-build
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: web
+          ref: ${{github.base_ref}}
+
+      - name: Report build failure
+        if: steps.master-build.outputs.conclusion == 'failure'
+        run: |
+          echo "Build failed on ${{github.base_ref}}"
+          exit 1
+
+      - name: Download build artifact from ${{github.base_ref}}
+        uses: dawidd6/action-download-artifact@v2
+        id: pr-build
+        with:
+          branch: ${{github.base_ref}}
+          workflow: build.yml
+          name: build-stats
+          path: base
+
+      - name: Download build artifact from PR
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏do we still need this step?

Reply:
Yes, artifacts aren’t automatically downloaded to the machine.